controllers/kustomization_controller.go

```diff
@@ -237,7 +237,7 @@ func (r *KustomizationReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	if reconcileErr != nil {
 		// record the reconciliation error
 		r.recordReadiness(ctx, reconciledKustomization)
-		return ctrl.Result{RequeueAfter: kustomization.Spec.Interval.Duration}, reconcileErr
+		return ctrl.Result{RequeueAfter: kustomization.Spec.Interval.Duration}, nil
```

Comment:
If we return an error here the resource will be immediately reconciled. So what is the point of setting RequeueAfter? Reconciling immediately will mean that the Kustomization will not reach a unhealthy condition until the backoff is large enough.

Reply:
Hmm something is very wrong if it doesn’t schedule at the RetryInteval. Please undo this change as it breaks the retry behavior.

Reply:
I am trying to find docs about the Reconcile behavior to be sure, but it feels like the requeue after is ignored if an error is returned. When a health check timeout error occurs the resources will reconcile again after 1 ms. As I understand it that is not what we want to happen?

Reply:
We want to retry the reconciliation based on what GetRetryInterval gives us. Maybe something changed in controller-runtime 0.8, is your PR based on the latest main branch?

Reply:
Yep i rebased from main yesterday. So it might be that or the in memory cluster not behaving like a "normal" cluster. I am seeing the same behavior in a real cluster where the condition will go from progressing to progressing as it is immediately rescheduled. Also the way we are patching status updates means that we will always override conditions in that status. The only way to determine if the health check timed out is to check the events.

Reply:
We need to introduce a new condition called heathCheck that records the last check result. But I would not mix this into this PR. First let’s figure out why the retry interval doesn’t work.

Reply:
I will investigate why it is reconciled immediately. So all we need to do is change the Type value in the status condition. The reason the conditions replaced is because the merge key for conditions is the Type value.

Reply:
The ready condition is reset by progressing and that’s ok, we need to create a dedicated condition for health checks.

Reply:
Looking into controller-runtime, if you return an error from a reconcile call the result will be ignored. This should be the right code that does the error handling.
https://github.com/kubernetes-sigs/controller-runtime/blob/9e78e653228851684b6b9cdabe5aae8559fe3722/pkg/internal/controller/controller.go#L297

I am not sure if it has always been like this or if it is a change introduced in the latest version. Either way we need to fix KC so that it actually waits the interval time.

Reply:
Fixed in #256, please rebase.